tests/test_hypothesis.py

```diff
@@ -95,6 +102,10 @@ def test_equality(d):
 @given(d=d, arg1=immutable, arg2=immutable, itemlist=lists(tuples(immutable, immutable), **sz))
 def test_consistency(arity, methodname, B, d, arg1, arg2, itemlist):
     ordered = issubclass(B, OrderedBidirectionalMapping)
+    if ordered and WORKAROUND_OD_NAN_BUG:
+        #assume(nan not in d and nan not in d.values())
+        # TODO: For some reason the assume does not work. Resort to:
+        d = {(k, v) for (k, v) in iteritems(d) if not isnan_(k) and not isnan_(v)}
```

Comment:
```pycon
>>> float('nan') in {float('nan')}
False
```

The behaviour with nan and collections containment is screwy. If you have two nans which are reference distinct (this can't happen on pypy but can on CPython) then they're distinct for contains.